plugins/github-pull-requests-board/src/hooks/useUserRepositoriesAndTeam.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef, useEntity } from '@backstage/plugin-catalog-react';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  getProjectNameFromEntity,
+  getGithubOrganizationFromEntity,
+  getUserNameFromEntity,
+} from '../utils/functions';
+
+export function useUserRepositoriesAndTeam() {
+  const { entity: teamEntity } = useEntity();
+  const catalogApi = useApi(catalogApiRef);
+  const [teamData, setTeamData] = useState<{
+    repositories: string[];
+    teamMembers: string[];
+  }>({
+    repositories: [],
+    teamMembers: [],
+  });
+
+  const getTeamData = useCallback(async () => {
+    // get team repositories and members
+    const entitiesList = await catalogApi.getEntities({
+      filter: [
+        { 'relations.ownedBy': stringifyEntityRef(teamEntity) },
+        { 'relations.memberOf': stringifyEntityRef(teamEntity) },
+      ],
+    });
+
+    const repositories = entitiesList.items.filter(
+      entity => entity.kind === 'Component',
+    );
+    const repositoriesNames: string[] = repositories.map(componentEntity =>
+      getProjectNameFromEntity(componentEntity),
+    );
+
+    const teamMembers = entitiesList.items.filter(
+      entity => entity.kind === 'User',
+    );
+    const teamMembersNames: string[] = teamMembers.map(componentEntity =>
+      getUserNameFromEntity(componentEntity),
+    );
+
+    setTeamData({
+      repositories: [...new Set(repositoriesNames)],
+      teamMembers: [...new Set(teamMembersNames)],
+    });
+  }, [catalogApi, teamEntity]);
+
+  useEffect(() => {
+    getTeamData();
+  }, [getTeamData]);
+
+  return {
```

Comment:
Would this benefit from a `loading` flag? just checking

As a matter of fact, the entire implementation above could be leveraging a `useAsync` if you'd like to, instead of the manual useEffect etc - this is all optional and feel free to ignore if you prefer the current structure.

Reply:
This file is the old `plugins/github-pull-requests-board/src/hooks/useUserRepositories.tsx` that I renamed and modified to include the team information loading.

The original implementation did not provide a `loading`, but I think it makes sense to add it, so I'll add that. It's probably not noticeable in most cases because the backstage backend API will return a very fast reply, and there is already some loading logic when querying the Github API, but it's likely helpful in the few cases where the Backstage API is not replying as fast as expected.

For the `useAsync`, I'm a real newbie when it comes to React, so I'd probably need more guidance on how to modify things from the existing `useEffect`, or I can keep it as it is